### ISSUE TITLE
Stacktrace returns Empty Frames

### ISSUE THREAD

**Issue by ajtran** [rollbar/rollbar-php#379]
Users have experienced empty frames in their stack trace.

Example:

```
{
  "body": {
    "trace": {
      "frames": [
        {}, 
        {}, 
        {}, 
        {}, 
        {}, 
        {}, 
        {}, 
        {}, 
        {}, 
        {}, 
        {}, 
        {}, 
        {}, 
        {}, 
        {}, 
        {}, 
        {}, 
        {}, 
        {}, 
        {}, 
        {}, 
        {}, 
        {}, 
        {}, 
        {}, 
        {}, 
        {}, 
        {}, 
        {}, 
        {}, 
        {}, 
        {}
      ],
```

**Comment by closca** [comment]
Hi,
I am facing the same issue. The last log which was ok is from 2018-06-13 13:19:48 EEST
Had 3 releases until the first log with the missing stack trace started to occur.

**Comment by ArturMoczulski** [comment]
This needs to be investigated. Added to the current milestone.

**Comment by alterphp** [comment]
RollbarLogger::encode(array &$payload) seems to be guilty.

In RollbarLogger::log($level, $toLog, array $context = array(), $isUncaught = false) method, from line 103 to 105, $scrubbed has the filled frames, $encoded has empty frames.

```php
            $scrubbed = $this->scrub($payload);
            $encoded = $this->encode($scrubbed);
            $truncated = $this->truncate($encoded);
```

**Comment by alterphp** [comment]
`RollbarLogger::encode(array &$payload)` does not push frames content into `$encoded` part.

**Comment by ArturMoczulski** [comment]
@alterphp yes, thank you. I'm working on a fix for this.

**Comment by alterphp** [comment]
A solution is to make Frame.php implementing JsonSerializable :
 
https://gist.github.com/alterphp/4c591440a52de008c15fa4e18637e09e.js

**Comment by ArturMoczulski** [comment]
We moved away from JsonSerializable. But yes the solution is down that line. Frame.php does not get encoded correctly. I should have a fix out within the next hour or so.

**Comment by alterphp** [comment]
ok, thanks

**Comment by alterphp** [comment]
@ArturMoczulski :+1: 

It's ok now ! Thanks for fast fix

**Comment by closca** [comment]
Is this included on v1.5.3 ?

**Comment by alterphp** [comment]
Yes !

Le mar. 19 juin 2018 à 12:46, Closca Costel <notifications@github.com> a
écrit :
